Add documentation on how to use Amazon Bedrock

srdas opened the pull request:
Address Issue #924 

Added docs to show how to use Amazon Bedrock models with Jupyter AI. Shows how to use both base models and custom models after fine-tuning, using the model `arn` (Amazon Resource Name). Results in an extra subsection in the User documentation: 
<img width="974" alt="image" src="https://github.com/user-attachments/assets/56bec887-716e-45f3-be72-1d5607c14eea">

The same section now contains a separate link at the end of the section that directs the user to a subpage with very detailed workflows on setting up and using Amazon Bedrock with Jupyter AI, including how to fine tune a based model to create a custom model and then call the custom model from Jupyter AI. See:
<img width="470" alt="image" src="https://github.com/user-attachments/assets/c40c104b-6dc6-4be1-b9e3-c3f84a426d9c">

Clicking through on this leads to the following web page: 
<img width="1054" alt="image" src="https://github.com/user-attachments/assets/f124ad65-7480-4497-8dee-3264c871015d">
A "Back" button is also provided to return to the main user documentation page as can be seen on the screenshot above.

srdas commented:
@JasonWeill Thanks for your comment ("I'm a bit concerned by how much we're adding that's specific to Amazon Bedrock, including screenshots of the console, instead of deferring to the AWS docs. I'm wondering if we could solicit a review from someone outside AWS, to get an impartial opinion about this"). 

I have shortened the main user documentation to include only usage in Jupyter AI, and linked it to a subpage where all Bedrock usage details are provided more extensively. Note the new screenshots above.

JasonWeill commented:
There are two one-sentence paragraphs near the big table of model providers that say, "To use the Bedrock models, you need access to the Bedrock service. For more information, see the [Amazon Bedrock Homepage](https://aws.amazon.com/bedrock/). … To use Bedrock models, you will need to authenticate via boto3." It might be good to merge these two into one, and to either add a sentence with a link to the Bedrock subpage to the or replace these with a link to the Bedrock subpage.

It might also be good to add a link to "Using Amazon Bedrock with Jupyter AI" to the _top_ of the "Amazon Bedrock Usage" section, either in addition to or instead of the link at the bottom.

The "<--Back" link doesn't look quite right, both because it doesn't use an actual arrow character like ← and because it sends the user back to a place where they may not have been — for example, if they landed on the "Using Amazon Bedrock with Jupyter AI" page from a search engine query. See #655 — this might be better suited for the curiously empty "Section Navigation" in the left pane, to display a hierarchy of documents.

srdas commented:
@JasonWeill Thanks so much for the comments. I have addressed them as follows: 

Edited the two one-sentence paragraphs down to a single one: 
![image](https://github.com/user-attachments/assets/fd96b595-6a26-4c28-a043-6d5b89e7a552)

Added an additional link to the top of the "Amazon Bedrock Usage" page just below the first introductory paragraph. 
![image](https://github.com/user-attachments/assets/3364ecb5-7dfe-4a43-82a5-8d9cfdc18f34)

Removed the vacant left panel that was unused. Addresses Issue #655 
![image](https://github.com/user-attachments/assets/19454747-9e04-4a71-90a4-def531368542)

You are right, the "Back" button is not meaningful and unclear as to where it leads. I replaced it with an explicit link as shown
![image](https://github.com/user-attachments/assets/9e772cbf-33fb-46e1-a63c-6acbe9cf41e0)

JasonWeill reviewed (approved):
Thank you! This is a good improvement.
